airflow/notifiers/__init__.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import jinja2
+import os
+
+from abc import ABCMeta, abstractmethod
+
+
+TEMPLATE_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'templates')
+
+
+class AirflowNotifier(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def send_notification(self, task_instance, state, message, **kwargs):
+        pass
+
+    def render_template(self, template, task_instance, **kwargs):
+        dag = task_instance.task.dag
+
+        jinja_loader = jinja2.FileSystemLoader([dag.folder, TEMPLATE_DIR])
```

Comment:
`FileSystemLoader` might be overkill here. It's great for complex `import/extends` logic but it might assume having to rebuild Airflow to alter a template. By that I mean that if you're defining a Notifier inline with your DAG or in your DAG repo somehow, you can't really mess with Airflow's template folder.

You could just have the template as a triple quoted string close to the Notifier class itself, and go without a jinja environment altogether